diff --git a/src/LinearMath/CMakeLists.txt b/src/LinearMath/CMakeLists.txt
index ecf06b3..7cbf430 100644
--- a/src/LinearMath/CMakeLists.txt
+++ b/src/LinearMath/CMakeLists.txt
@@ -57,6 +57,7 @@ SET(LinearMath_HDRS
 ADD_LIBRARY(LinearMath ${LinearMath_SRCS} ${LinearMath_HDRS})
 SET_TARGET_PROPERTIES(LinearMath PROPERTIES VERSION ${BULLET_VERSION})
 SET_TARGET_PROPERTIES(LinearMath PROPERTIES SOVERSION ${BULLET_VERSION})
+target_compile_definitions(LinearMath PUBLIC BT_ENABLE_PROFILE)
 
 IF (INSTALL_LIBS)
 	IF (NOT INTERNAL_CREATE_DISTRIBUTABLE_MSVC_PROJECTFILES)
