From 84dc6af45b1176a2b82d089239665cb3dc1584de Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20Gro=C3=9F?= <firedefender1@googlemail.com>
Date: Wed, 23 Dec 2020 07:33:39 +0100
Subject: [PATCH] Hack: remove #include <sys/sysctl.h> to build on
 linux-aarch64

---
 src/nvthread/nvthread.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/nvthread/nvthread.cpp b/src/nvthread/nvthread.cpp
index 2b3abe6..cec9f4f 100644
--- a/src/nvthread/nvthread.cpp
+++ b/src/nvthread/nvthread.cpp
@@ -11,7 +11,6 @@
 #include <unistd.h>
 #elif NV_OS_UNIX
 #include <sys/types.h>
-#include <sys/sysctl.h>
 #include <unistd.h>
 #elif NV_OS_DARWIN
 #import <stdio.h>
-- 
2.17.1

