diff --git a/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.cpp b/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.cpp
index ad87cb0..54ed18a 100644
--- a/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.cpp
+++ b/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.cpp
@@ -117,6 +117,8 @@ QOpenSLESAudioInput::QOpenSLESAudioInput(const QByteArray &device)
         m_recorderPreset = SL_ANDROID_RECORDING_PRESET_CAMCORDER;
     else if (qstrcmp(device, QT_ANDROID_PRESET_VOICE_RECOGNITION) == 0)
         m_recorderPreset = SL_ANDROID_RECORDING_PRESET_VOICE_RECOGNITION;
+    else if (qstrcmp(device, QT_ANDROID_PRESET_VOICE_COMMUNICATION) == 0)
+        m_recorderPreset = SL_ANDROID_RECORDING_PRESET_VOICE_COMMUNICATION;
     else
         m_recorderPreset = SL_ANDROID_RECORDING_PRESET_GENERIC;
 #endif
diff --git a/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.h b/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.h
index ad84db0..35cc379 100644
--- a/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.h
+++ b/qtmultimedia/src/plugins/opensles/qopenslesaudioinput.h
@@ -50,6 +50,7 @@
 #define QT_ANDROID_PRESET_MIC "mic"
 #define QT_ANDROID_PRESET_CAMCORDER "camcorder"
 #define QT_ANDROID_PRESET_VOICE_RECOGNITION "voicerecognition"
+#define QT_ANDROID_PRESET_VOICE_COMMUNICATION "voicecommunication"
 
 #endif
 
diff --git a/qtmultimedia/src/plugins/opensles/qopenslesengine.cpp b/qtmultimedia/src/plugins/opensles/qopenslesengine.cpp
index 1a16cc2..2577fb3 100644
--- a/qtmultimedia/src/plugins/opensles/qopenslesengine.cpp
+++ b/qtmultimedia/src/plugins/opensles/qopenslesengine.cpp
@@ -114,7 +114,8 @@ QList<QByteArray> QOpenSLESEngine::availableDevices(QAudio::Mode mode) const
 #ifdef ANDROID
         devices << QT_ANDROID_PRESET_MIC
                 << QT_ANDROID_PRESET_CAMCORDER
-                << QT_ANDROID_PRESET_VOICE_RECOGNITION;
+                << QT_ANDROID_PRESET_VOICE_RECOGNITION
+                << QT_ANDROID_PRESET_VOICE_COMMUNICATION;
 #else
         devices << "default";
 #endif
 