diff --git a/qtscript/src/script/api/qscriptengine.cpp b/qtscript/src/script/api/qscriptengine.cpp
index c5f437b..942833e 100644
--- a/qtscript/src/script/api/qscriptengine.cpp
+++ b/qtscript/src/script/api/qscriptengine.cpp
@@ -1332,8 +1332,10 @@ void QScriptEnginePrivate::collectGarbage()
 
 void QScriptEnginePrivate::reportAdditionalMemoryCost(int size)
 {
-    if (size > 0)
+    if (size > 0) {
+        QScript::APIShim shim(this);
         globalData->heap.reportExtraMemoryCost(size);
+    }
 }
 
 QScript::TimeoutCheckerProxy *QScriptEnginePrivate::timeoutChecker() const
